azurerm/resource_arm_batch_certificate.go

```diff
@@ -0,0 +1,256 @@
+		d.Set("thumbprint_algorithm", props.ThumbprintAlgorithm)
+	}
+
+	return nil
+}
+
+func resourceArmBatchCertificateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).batchCertificateClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure Batch certificate update.")
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	accountName := id.Path["batchAccounts"]
+	name := id.Path["certificates"]
+	resourceGroupName := id.ResourceGroup
+
+	certificate := d.Get("certificate").(string)
+	format := d.Get("format").(string)
+	password := d.Get("password").(string)
+	thumbprint := d.Get("thumbprint").(string)
+	thumbprintAlgorithm := d.Get("thumbprint_algorithm").(string)
+
+	if err := validateFormatAndPassword(format, password); err != nil {
+		return err
+	}
+
+	parameters := batch.CertificateCreateOrUpdateParameters{
+		Name: &name,
+		CertificateCreateOrUpdateProperties: &batch.CertificateCreateOrUpdateProperties{
+			Data:                &certificate,
+			Format:              batch.CertificateFormat(format),
+			Password:            &password,
+			Thumbprint:          &thumbprint,
+			ThumbprintAlgorithm: &thumbprintAlgorithm,
+		},
+	}
+
+	if _, err = client.Update(ctx, resourceGroupName, accountName, name, parameters, ""); err != nil {
+		return fmt.Errorf("Error updating Batch certificate %q (Account %q / Resource Group %q): %+v", name, accountName, resourceGroupName, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroupName, accountName, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Batch Certificate %q (Account %q / Resource Group %q): %+v", name, accountName, resourceGroupName, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read ID for Batch certificate %q (Account: %q, Resource Group %q) ID", name, accountName, resourceGroupName)
+	}
+
+	return resourceArmBatchCertificateRead(d, meta)
+}
+
+func resourceArmBatchCertificateDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).batchCertificateClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	accountName := id.Path["batchAccounts"]
+	name := id.Path["certificates"]
+	resourceGroupName := id.ResourceGroup
+
+	future, err := client.Delete(ctx, resourceGroupName, accountName, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting Batch Certificate %q (Account %q / Resource Group %q): %+v", name, accountName, resourceGroupName, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		if !response.WasNotFound(future.Response()) {
+			return fmt.Errorf("Error waiting for deletion of Batch Certificate %q (Account %q / Resource Group %q): %+v", name, accountName, resourceGroupName, err)
+		}
+	}
+
+	return nil
+}
+
+func validateFormatAndPassword(format string, password string) error {
```

Comment:
in Go private methods are shared throughout the entire Package, as such could we prefix this method name e.g.

```suggestion
func validateBatchCertificateFormatAndPassword(format string, password string) error {
```